### ISSUE TITLE
Grains host/hostname documentation

### ISSUE THREAD

**Issue by dwickwire** [saltstack/salt#41174]
### Description of Issue/Question

The [docs](https://docs.saltstack.com/en/latest/ref/grains/all/salt.grains.core.html#salt.grains.core.hostname) say to use hostname but when I use {{ grains.hostname }} I get nothing, if I use `{{ grains['hostname'] }}` I get a KeyError.  

However after doing some looking through the grains core code I saw `host` is populated and thus `{{ grains['host'] }}` was available in a state.  

- Is the core not the correct place to look for this kind of documentation andor should it be labeled host and not hostname?  
- How would one use `hostname` in the state?

### Setup
```yml
hostname: {{ grains['hostname'] }} # doesn't work
hostname: {{ grains['host'] }} # works
```
### Error

```shell
[ERROR   ] Rendering exception occurred: Jinja variable 'dict object' has no attribute 'hostname'
[ERROR   ] Unable to manage file: Jinja variable 'dict object' has no attribute 'hostname'
local:
----------
          ID: /etc/cron.d/mycron
    Function: file.managed
      Result: False
     Comment: Unable to manage file: Jinja variable 'dict object' has no attribute 'hostname'
     Started: 18:14:31.035735
    Duration: 54.701 ms
     Changes:   
```

### Versions Report
Version: 2016.11.3

**Comment by sebw** [comment]
By looking at the code salt.grains.core.hostname() returns grains host, fqdn, localhost and domain:

```
def hostname():
    '''
    Return fqdn, hostname, domainname
    '''
    # This is going to need some work
    # Provides:
    #   fqdn
    #   host
    #   localhost
    #   domain
    global __FQDN__
    grains = {}

    if salt.utils.is_proxy():
        return grains

    grains['localhost'] = socket.gethostname()
    if __FQDN__ is None:
        __FQDN__ = salt.utils.network.get_fqhostname()
    grains['fqdn'] = __FQDN__
    (grains['host'], grains['domain']) = grains['fqdn'].partition('.')[::2]
    return grains
```

**Comment by dwickwire** [comment]
Gotcha, then this is my misunderstanding.  Thanks.

**Comment by akaii78** [comment]
what is the aim you are looking to do with {{ grains.hostname }}?

**Comment by dwickwire** [comment]
We use it to distinguish which cron entries get provisioned for that specific server.  We have backup servers to switch to which are also the same production environment but we don't want cron tasks running on them until we actually switch.

I closed this because I was able to use `grains['host']` which I found by looking at @sebw suggested command.   I could probably use a different pillar but that would be more complicated to switch my entire state structure for one small area of difference.
